rewriter/TEnum.cc

```diff
@@ -118,6 +119,12 @@ std::optional<ProcessStatResult> processStat(core::MutableContext ctx, ast::Clas
 
     auto *selfNew = findSelfNew(asgn->rhs);
     if (selfNew == nullptr) {
+        if (fromWhere == FromWhere::After) {
+            // Allow non-enum constants to be defined after `enums do`
+            // (type aliases, type members, etc.)
```

Comment:
Technically you can define a type member, but at the moment it won't work well because there will be no way to redeclare that type member inside the fake singleton classes which we create for each enum value.

We could support this in a future change (and probably will in service of #4035 when we get around to it).